# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../tcp_wrappers/0026-tcp_wrappers-7.6-size_t.patch
# Copyright (C) 2011 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

diff -ruN tcp_wrappers_7.6.orig/fix_options.c tcp_wrappers_7.6/fix_options.c
--- tcp_wrappers_7.6.orig/fix_options.c	2003-08-21 03:41:33.000000000 +0200
+++ tcp_wrappers_7.6/fix_options.c	2003-08-21 03:41:27.000000000 +0200
@@ -38,7 +38,11 @@
 #ifdef IP_OPTIONS
     unsigned char optbuf[BUFFER_SIZE / 3], *cp;
     char    lbuf[BUFFER_SIZE], *lp;
+#ifdef __GLIBC__
+    size_t  optsize = sizeof(optbuf), ipproto;
+#else
     int     optsize = sizeof(optbuf), ipproto;
+#endif
     struct protoent *ip;
     int     fd = request->fd;
     unsigned int opt;
diff -ruN tcp_wrappers_7.6.orig/socket.c tcp_wrappers_7.6/socket.c
--- tcp_wrappers_7.6.orig/socket.c	2003-08-21 03:41:33.000000000 +0200
+++ tcp_wrappers_7.6/socket.c	2003-08-21 03:40:51.000000000 +0200
@@ -90,7 +90,11 @@
     static struct sockaddr_in client;
     static struct sockaddr_in server;
 #endif
+#ifdef __GLIBC__
+    size_t  len;
+#else
     int     len;
+#endif
     char    buf[BUFSIZ];
     int     fd = request->fd;
 
@@ -421,7 +425,11 @@
 #else
     struct sockaddr_in sin;
 #endif
+#ifdef __GLIBC__
+    size_t  size = sizeof(sin);
+#else
     int     size = sizeof(sin);
+#endif
 
     /*
      * Eat up the not-yet received datagram. Some systems insist on a
